### ISSUE TITLE
Multiple/no value params give weird results

### ISSUE THREAD

**Issue by Petah** [thephpleague/url#58]
Given this URL:

http://localhost?foo=1&foo=2&bar

The query is:
```php
 protected 'query' => 
    object(League\Url\Components\Query)[24]
      protected 'data' => 
        array (size=2)
          'bar&foo' => string '1' (length=1)
          'foo' => string '2' (length=1)
```

**Comment by nyamsprod** [comment]
Strange... What version of `League\Url` and `PHP` are you using ?
here's the output I got with version 3.2.1 in PHP5.5:
```php
$url = League\Url\Url::createFromUrl('http://localhost?foo=1&foo=2&bar');
var_export($url->getQuery()->toArray());
```
output the following:

```php
array (
  'foo' => '2',
)
```
The fact that `bar` is missing is debatable and will surely be fixed in v4. But I don't get `bar&foo` in the output

**Comment by nyamsprod** [comment]
A new 3.x version is out with the bug fix about the missing bar element

**Comment by Petah** [comment]
Thanks for that, turns out it was actually Symfony Request causing the `foo&bar` issue.

**Comment by Petah** [comment]
Sorry I was mistaken. Using 3.3.0, given this URL: 

```
$url = 'http://localhost:82/url.php?foo&bar&baz';
```

I would expect:
```
array (size=3)
  'foo' => string '' (length=0)
  'bar' => string '' (length=0)
  'baz' => string '' (length=0)
```

But I get:
```
array (size=1)
  'foo&bar&baz' => string '' (length=0)
```

**Comment by Petah** [comment]
It seems this is what is causing it:

https://github.com/thephpleague/url/blob/1c1fb645b5a7de3d7e7bef07e203ec37f6b34847/src/Components/Query.php#L118-L120

Not sure why you are even doing that.

**Comment by nyamsprod** [comment]
this is done because  PHP's `parse_str` convert `.` into `_` in query parameter names see #31 for more info.

**Comment by nyamsprod** [comment]
Hey thanks for the fix I'll review it but in the meantime I've also resolve the bug by fixing the regular expression. I'll compare both solutions to select which is better/faster and less error prone.

**Comment by nyamsprod** [comment]
Whatever the solution I choose in the end I think a bug fix version will be out this week :+1:

**Comment by nyamsprod** [comment]
a bug fix version has been released with the fix behavior

**Comment by strelec** [comment]
This bug is caused because you had a wrong regex:
```
/(?:^|(?<=&))[^=[]+/
```
The correct one is:
```
/(^|(?<=&))[^=[&]+/
```

**Comment by strelec** [comment]
The code that works on PHP below 5.4 can be seen here:

http://stackoverflow.com/a/18209799/924109

**Comment by Petah** [comment]
Modern libraries should not be supporting below PHP 5.4

Even PHP 5.4 should only be supported for security fixes now, and EOL in 1 month.

**Comment by nyamsprod** [comment]
@Petah league url v3 was released when PHP 5.3 was not yet EOL. League url v4 development was started being PHP5.4 only. Because development took much more time league uri v4 will be PHP 5.5 only hope this clarify the situation.

**Comment by Petah** [comment]
Fair enough :) good lib BTW.
